eventsourcing/domain/services/transcoding.py

```diff
@@ -35,122 +33,103 @@ def deserialize(self, stored_event):
 
 
 class Transcoder(AbstractTranscoder):
+    """
+    default implementation of a Transcoder
+    """
 
     serialize_without_json = False
     serialize_with_uuid1 = True
 
+    StoredEvent = namedtuple('StoredEvent', ['event_id', 'stored_entity_id', 'event_topic', 'event_attrs'])
+
     def __init__(self, json_encoder_cls=None, json_decoder_cls=None, cipher=None, always_encrypt=False):
         self.json_encoder_cls = json_encoder_cls
         self.json_decoder_cls = json_decoder_cls
         self.cipher = cipher
         self.always_encrypt = always_encrypt
 
     def serialize(self, domain_event):
-        return serialize_domain_event(
-            domain_event,
-            json_encoder_cls=self.json_encoder_cls,
-            without_json=self.serialize_without_json,
-            with_uuid1=self.serialize_with_uuid1,
-            cipher=self.cipher,
-            always_encrypt=self.always_encrypt,
+        """
+        Serializes a domain event into a stored event.
+        """
+        # assert isinstance(domain_event, DomainEvent)
+
+        # Copy the state of the domain event.
+        event_attrs = domain_event.__dict__.copy()
+
+        # Get, or make, the domain event ID.
+        if self.serialize_with_uuid1:
+            event_id = event_attrs.pop('domain_event_id')
+        else:
+            event_id = uuid.uuid4().hex
+
+        # Make stored entity ID and topic.
+        stored_entity_id = make_stored_entity_id(id_prefix_from_event(domain_event), domain_event.entity_id)
+        event_topic = topic_from_domain_class(type(domain_event))
+
+        # Serialise event attributes to JSON, optionally encrypted with cipher.
+        if not self.serialize_without_json:
+
+            if self.json_encoder_cls is None:
+                self.json_encoder_cls = ObjectJSONEncoder
+
+            event_attrs = json.dumps(event_attrs, separators=(',', ':'), sort_keys=True, cls=self.json_encoder_cls)
+
+            if self.always_encrypt or domain_event.__class__.always_encrypt:
+                if self.cipher is None:
+                    raise ValueError("Can't encrypt without a cipher")
+                event_attrs = self.cipher.encrypt(event_attrs)
+
+        # Return a named tuple.
+        return self.StoredEvent(
+            event_id=event_id,
+            stored_entity_id=stored_entity_id,
+            event_topic=event_topic,
+            event_attrs=event_attrs,
         )
```

Comment:
Ah I also did this. I should've start on the archived logs :-). Don't worry, I'll take your changes.